application/src/main/java/org/togetherjava/tjbot/commands/free/FreeCommand.java

```diff
@@ -0,0 +1,330 @@
+            // approves
+            channelMonitor.setChannelFree(id);
+            displayStatus(channelMonitor.getStatusChannelFor(event.getGuild()));
+            event.reply(UserStrings.MARK_AS_FREE.message()).queue();
+        } else {
+            Util.sendErrorMessage(event, UserStrings.ALREADY_FREE_ERROR.message());
+        }
+    }
+
+    /**
+     * Private method to test event to see if it should be processed.
+     *
+     * Will respond to users describing the problem if the event should not be processed.
+     *
+     * @param event the event to test for validity.
+     * @return true if the event should be processed false otherwise.
+     */
+    private boolean shouldHandle(@NotNull final SlashCommandEvent event) {
+        if (!isReady) {
+            logger.debug(
+                    "Slash command requested by {} in {}(channel: {}) before command is ready.",
+                    event.getUser().getIdLong(), event.getGuild(), event.getChannel().getName());
+            Util.sendErrorMessage(event, UserStrings.NOT_READY_ERROR.message());
+            return false;
+        }
+        if (!channelMonitor.isMonitoringGuild(event.getGuild().getIdLong())) {
+            logger.error(
+                    "Slash command used by {} in {}(channel: {}) when guild is not configured for Free Command",
+                    event.getUser().getIdLong(), event.getGuild(), event.getChannel().getName());
+            Util.sendErrorMessage(event,
+                    UserStrings.NOT_CONFIGURED_ERROR.formatted(event.getGuild().getName()));
+            return false;
+        }
+        if (!channelMonitor.isMonitoringChannel(event.getChannel().getIdLong())) {
+            logger.debug("'/free called in un-configured channel {}({})",
+                    event.getGuild().getName(), event.getChannel().getName());
+            Util.sendErrorMessage(event, UserStrings.NOT_MONITORED_ERROR.message());
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Builds the message that will be displayed for users.
+     * <p>
+     * This method dynamically builds the status message as per the current values on the guild,
+     * including the channel categories. This method will detect any changes made on the guild and
+     * represent those changes in the status message.
+     *
+     * @param channel the text channel the status message will be posted in.
+     */
+    public void displayStatus(@NotNull TextChannel channel) {
+        final Guild guild = channel.getGuild();
+
+        String messageTxt = buildStatusMessage(guild);
+        MessageEmbed embed = MessageUtils.generateEmbed(STATUS_TITLE, messageTxt,
+                channel.getJDA().getSelfUser(), FREE_COLOR);
+
+        long latestMessageId = channel.getLatestMessageIdLong();
+        Optional<Message> statusMessage = getStatusMessageIn(channel);
+        if (statusMessage.isPresent()) {
+            Message message = statusMessage.get();
```

Comment:
i dont want to throw
and i dont want to have to try-catch so that i can throw

 not having a value is not an error
it just needs to be handled differantly

Reply:
`get()` does throw. its 1:1 identical with `orElseThrow()` with the difference that the naming on latter is better.

Reply:
Im not saying you should replace ur `if` check. Just that you should literally replace `statusMessage.get()` by `statusMessage.orElseThrow()`.

Reply:
fixed